mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,32 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = Object.hasOwn(this.constructor.prototype, 'getLoadingComplete')
+		? new Promise(resolve => this.#loadingCompleteResolve = resolve)
+		: Promise.resolve();
+
+	get loadingComplete() {
+		return this.getLoadingComplete();
 	}
 
-	getLoadingComplete() {
-		return this._loadingCompletePromise;
+	get resolveLoadingComplete() {
+		return () => this.#resolveLoadingComplete();
 	}
```

Comment:
Without the bind this has me tempted to suggest dropping the `#resolveLoadingComplete` body directly into the arrow definition just so we can strip out that extra layer of indirection, it's not being used elsewhere

Reply:
Yes! +1 arrow functions